File: test/11.6.0/AES/simple/prog.c
#include    "/Users/dun202/install/3pl/include/cpu_serial.h"
#include    <fts.h>
#include    <math.h>



static char     usbsearch[] = "tty.usbmodem";   // for AVnet board on OSX
static int      usbsearchlen = 12;
static char     dev[6] = "/dev/";
char            portName[200];

int         compare (const FTSENT**, const FTSENT**);
void        findserialdev ();

#include    "prog_fpga.h"

int
main (
    int     argc,
    char**  argv
) {    
    
    // Find the interface serial device.
    findserialdev();
    
    // Establish the CPU interface.
    open_cpu_serial(portName, INTERRUPTS);


    fpga_LW_write(7);

    fpga_S1W_write(47);
    printf("%d\n", fpga_S2R_read());

  
    // Close the CPU interface.
    close_cpu_serial();

    exit(0);
}


// Search for a serial port.
// 'usbsearch' gives the head of the device name and 'usbsearchlen' is the
// length of string to search.
// IF MORE THAN ONE PORT MATCHES THE LAST IS USED.
void
findserialdev() {
    FTS*    file_system = NULL;
    FTSENT* child = NULL;
    FTSENT* parent = NULL;
    char*   pathlist[2] = {"/dev", NULL};

    file_system = fts_open(pathlist, FTS_COMFOLLOW | FTS_NOCHDIR, &compare);

    if (NULL != file_system) {
        while( (parent = fts_read(file_system)) != NULL) {
            child = fts_children(file_system, 0);

            if (errno != 0)
                perror("error searching /dev for serial interface device");
            while ((NULL != child) && (NULL != child->fts_link)) {
                child = child->fts_link;
                if (strncmp(child->fts_name, usbsearch, usbsearchlen) == 0) {
                    strcpy(portName, dev);
                    strcat(portName, child->fts_name);
                    printf("found port = %s\n", portName);
                }
            }
        }
        if (strlen(portName) == 0) {
            printf("unable to find serial port\n");
            exit(1);
        }
        fts_close(file_system);
    }
}

int compare(const FTSENT** one, const FTSENT** two) {
    return (strcmp((*one)->fts_name, (*two)->fts_name));
}

File: test/11.6.0/AES/struct1/prog.py
#! /usr/bin/env python3

import sys
sys.path.append('/Users/dun202/install/include')
import cpu_serial
import prog_if


#portName = findserialdev("tty.usbmodem")
portName = "/dev/tty.usbmodem14201"
cpu_serial.open_cpu_serial(portName, 0)

vs = prog_if.fpga_SR_read()
print("f1 " + str(vs['f1']) + " (511)");
print("f2 " + str(vs['f2']) + " (2047)");
print("f3 " + str(vs['f3']) + " (True)");
print("f4 " + str(vs['f4']) + " (32767)");

vs['f1'] = 321
vs['f2'] = -1234
vs['f3'] = False
vs['f4'] = 54321
prog_if.fpga_SW_write(vs)

vs = prog_if.fpga_SR_read()
print("f1 " + str(vs['f1']) + " (321)");
print("f2 " + str(vs['f2']) + " (-1234)");
print("f3 " + str(vs['f3']) + " (False)");
print("f4 " + str(vs['f4']) + " (54321)");

cpu_serial.close_cpu_serial()

exit(0)
